frontend/src/scenes/authentication/redirectToLoggedInInstance.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * This will redirect the user to the correct subdomain if they're logged in to a different instance.
+ *
+ * It will only work for app. and eu. subdomains and other subdomains will need to be added manually
+ *
+ * There's an e2e test for this.
+ *
+ * ## Local testing
+ *
+ * To test this locally, you can edit your /etc/hosts to add a subdomain redirect
+ *
+ * Add the following line to your host file:
+ * 127.0.0.1 app.posthogtest.com
+ *
+ * Then set the following cookies locally:
+ * document.cookie = "ph_current_instance=eu.posthog.com";
+ * document.cookie = "is-logged-in=1";
+ *
+ * Then go to http://app.posthogtest.com:8000/login?next=/apps
+ * And it will update the subdomain, taking you to the following link
+ * http://eu.posthogtest.com:8000/login?next=/apps
+ */
+
+import { lemonToast } from '@posthog/lemon-ui'
+import { getCookie } from 'lib/api'
+
+// cookie values
+const PH_CURRENT_INSTANCE = 'ph_current_instance'
+const IS_LOGGED_IN = 'is-logged-in'
+
+const REDIRECT_TIMEOUT = 2500
+
+const SUBDOMAIN_TO_NAME = {
+    eu: 'EU cloud',
+    app: 'US cloud',
+}
```

Comment:
```suggestion
} as const
```
for some slightly nicer typing